docs/system_overview/configuration.md

```diff
@@ -14,254 +14,676 @@ This scrambled password file is saved after an `iinit` is run. If this file does
-  - `zone_name` (required) - The name of the Zone in which the server participates. This must be the same across all iRODS servers in a Zone.
-  - `zone_port` (required) (default 1247) - The main port used by the Zone for communication. This must be the same across all iRODS servers in a Zone.
-  - `zone_user` (required) - The name of the rodsadmin user running this iRODS instance.
-
-
-Servers in the catalog provider role have the following additional top level entries related to the database connection:
-
-  - `catalog_database_type` (required) - The type of database iRODS is using for the iCAT, either 'postgres', 'mysql', or 'oracle'
-
-  - `db_host` (required) - The hostname of the database server (can be localhost)
+```json
+{
+    // TODO
+    // Contains subtle network and password related variables.
+    // These values should be changed only in concert with all connecting clients and other servers in the Zone.
+    "advanced_settings": {
+        // The number of threads enabled when parallel transfer is invoked.
+        "default_number_of_transfer_threads": 4,
+
+        // The number of seconds a server-side temporary password is good.
+        "default_temporary_password_lifetime_in_seconds": 120,
+
+        // (Optional)
+        // A list of iRODS server hostnames the delay server dispatches rules to for execution.
+        // If the list is empty, the delay server will dispatch rules to the local server for execution.
+        "delay_rule_executors": [],
+
+        // (Optional)
+        // The number of seconds the delay server sleeps before checking the catalog for rules
+        // ready for execution.
+        "delay_server_sleep_time_in_seconds": 30,
+
+        // (Optional)
+        // Contains settings for controlling the behavior of the internal DNS cache.
+        "dns_cache": {
+            // The amount of shared memory allocated to the DNS cache.
+            "shared_memory_size_in_bytes": 5000000,
+
+            // The amount of time an entry stays valid before being evicted from the cache.
+            "eviction_age_in_seconds": 3600
+        },
 
-  - `db_odbc_type` (required) - The ODBC type, usually 'unix'
+        // (Optional)
+        // Contains settings for controlling the behavior of the internal Hostname cache.
+        "hostname_cache": {
+            // The amount of shared memory allocated to the Hostname cache.
+            "shared_memory_size_in_bytes": 2500000,
 
-  - `db_password` (required) - The password for the `db_username` to connect to the `db_name`
+            // The amount of time an entry stays valid before being evicted from the cache.
+            "eviction_age_in_seconds": 3600
+        },
 
-  - `db_port` (required) - The port on which the database server is listening
+        // TODO
+        "maximum_size_for_single_buffer_in_megabytes": 32,
```

Comment:
Anyone know what this is?

Is it only used for put operations?

Reply:
not just put - any single buffer transfers (put/get/repl).